oeps/architectural-decisions/oep-0065-frontend-composability.rst

```diff
@@ -0,0 +1,492 @@
+OEP-65: Frontend Composability
+##############################
+
+.. list-table::
+   :widths: 25 75
+
+   * - OEP
+     - :doc:`OEP-65 <oep-0065-arch-frontend-composability>`
+   * - Title
+     - Frontend Composability
+   * - Last Modified
+     - 2024-04-12
+   * - Authors
+     -
+       * Adolfo R. Brandes <adolfo@axim.org>
+       * Pedro Martello <pedro@hammerlabs.net>
+       * David Joy <david@joy.engineering>
+   * - Arbiter
+     - Adam Stankiewicz <astankiewicz@2u.com>
+   * - Status
+     - Under Review
+   * - Type
+     - Architecture
+   * - Created
+     - 2024-04-03
+   * - Review Period
+     - April 15, 2024 - April 29, 2024
+   * - Resolution
+     - <links to any discussions where the final status was decided>
+   * - References
+     -
+       * `FC-0054 - Composable Micro-frontends Discovery <https://openedx.atlassian.net/wiki/spaces/COMM/pages/4063821827/FC-0054+-+Composable+Micro-frontends+Piral+Discovery>`_
+       * `FC-0007 - Modular MFE Domains Discovery <https://openedx.atlassian.net/wiki/spaces/COMM/pages/3614900241/CLOSED+FC-0007+-+Modular+MFE+Domains+Discovery>`_
+
+
+Abstract
+********
+
+This OEP proposes that Open edX micro-frontends (MFEs) adopt module federation as an approach to address inconsistencies, performance issues, and inflexibilities inherent in the current MFE architecture.
+
+Motivation
+**********
+
+Micro-frontends were originally designed to avoid some of the limitations of the monolithic ``edx-platform`` frontend, namely that otherwise independent teams were beholden to the build, test, and release lifecycle of the rest of the codebase. This dramatically slowed down the pace of frontend feature development, experimentation, and innovation.
```

Comment:
Cautious nit: Is it worth clarifying that this was done explicitly for the benefits of those deploying off of master? AFAIK there's no particular benefit if you're strictly running off a named release, beyond more operational headache. There may be some benefit if you modify a given named release's code?

Reply:
I can see benefits to not having a monolith even when running a named release. For example: if someone were to fork off `quince.1` and modify a single MFE, they could still pull in `quince.2` and `quince.3` for the MFEs that they did not modify.

Reply:
I would say the strongest benefit is felt by instances so large that actually have multiple independent teams. The most well known large instance just so happens to have both situations in parallel.
Still I think it is work clarifying that.

Reply:
I'm adding some detail that it was: 1) for independent teams in edX's large engineering org, and 2) that it primarily benefits the master branch where the rate of change is particularly high.

Reply:
OK, thanks. I agree... decision docs of _why_ we went with MFEs would be nice...